demo-app/src/androidTest/java/com/worldpay/access/checkout/sample/card/standard/testutil/NewCardFragmentTestUtils.kt

```diff
@@ -0,0 +1,197 @@
+        wait2 { onView(withId(R.id.card_flow_payments_cvc_switch)).check(matches(isDisplayed())) }
+        wait2 { onView(withId(R.id.card_flow_payments_cvc_switch)).check(matches(checkedViewMatcher)) }
+    }
+
+    fun requestIsInProgress() = apply {
+        progressBarIsVisible()
+        enabledStateIs(pan = false, cvc = false, expiryDate = false, paymentsCvcSwitch = false, submitButton = false)
+    }
+
+    fun hasResponseDialogWithMessage(response: String) = apply {
+        dialogHasText(response)
+    }
+
+    fun hasErrorDialogWithMessage(error: String) = apply {
+        dialogHasText(error)
+    }
+
+    fun closeDialog() = apply {
+        onView(withId(android.R.id.button1)).perform(click())
+    }
+
+    fun isInErrorState(pan: String? = null, cvc: String? = null, expiryDate: String? = null) = apply {
+        progressBarNotVisible()
+        enabledStateIs(pan = true, cvc = true, expiryDate = true, paymentsCvcSwitch = true, submitButton = true)
+        cardDetailsAre(pan, cvc, expiryDate)
+    }
+
+    fun enabledStateIs(
+        pan: Boolean? = null,
+        cvc: Boolean? = null,
+        expiryDate: Boolean? = null,
+        paymentsCvcSwitch: Boolean? = null,
+        submitButton: Boolean? = null
+    ) = apply {
+        lateinit var enabledViewMatcher: Matcher<View>
+        lateinit var viewMatcher: Matcher<View>
+
+        val accessCheckoutFieldViewMatcher: AccessCheckoutFieldViewMatcher = when {
+            pan != null -> AccessCheckoutFieldViewMatcher.PanViewMatcher(pan)
+            cvc != null -> AccessCheckoutFieldViewMatcher.CvcViewMatcher(cvc)
+            expiryDate != null -> AccessCheckoutFieldViewMatcher.ExpiryDateViewMatcher(expiryDate)
+            paymentsCvcSwitch != null -> AccessCheckoutFieldViewMatcher.SwitchViewMatcher(paymentsCvcSwitch)
+            submitButton != null -> AccessCheckoutFieldViewMatcher.SubmitButtonViewMatcher(submitButton)
+            else -> throw RuntimeException("field view matcher not recognised")
+        }
+
+        wait2 { onView(accessCheckoutFieldViewMatcher.viewMatcher).check(matches(isDisplayed())) }
+        wait2 { onView(accessCheckoutFieldViewMatcher.viewMatcher).check(matches(accessCheckoutFieldViewMatcher.enabledViewMatcher)) }
+
+    }
+
+    fun clickSubmitButton() = apply {
+        enabledStateIs(submitButton = true)
+        uiObjectWithId(R.id.card_flow_btn_submit).click()
+    }
+
+    fun focusOn(input: Input) = apply {
+        when (input) {
+            Input.PAN -> uiObjectWithId(R.id.card_flow_text_pan).click()
+            Input.CVC -> uiObjectWithId(R.id.cvc_flow_text_cvc).click()
+            Input.EXPIRY_DATE -> uiObjectWithId(R.id.card_flow_expiry_date).click()
+        }
+    }
+
+    fun enterCardDetails(
+        pan: String? = null,
+        cvc: String? = null,
+        expiryDate: String? = null,
+        assertText: Boolean = false
+    ) = apply {
+        if (pan != null) enterText(R.id.card_flow_text_pan, pan)
```

Comment:
FYI you can also use the following syntax for executing a statement if something is not null

```
value?.let {
    ... // execute this block if not null
}
```

Just FYI, you don't need to change it here